### PR TITLE
deduplicate package names in `check_installs()` output

### LINKED ISSUE #723: duplicated package names in `check_installs()` output
```r
tune_grid(
  nearest_neighbor("regression", neighbors = tune()),
  mpg ~ .,
  bootstraps(mtcars)
)
#> Error in `check_installs()`:
#> ! Some package installs are required: 
#> • 'kknn', 'kknn'
```

### PR DESCRIPTION
**Pull request by simonpcouch**
To do so properly, switches to cli under the hood, adds tests, and routes the caller env through needed machinery.

Closes #723. :)

### CODE REVIEW

File: tests/testthat/test-checks.R [hunk 1]
```diff
@@ -179,6 +179,23 @@ test_that("workflow objects", {
   })
 })
 
+test_that("errors informatively with needed package install", {
+  # rstanarm is not installed during CI runs
+  # in contexts where it _is_ installed, skip the test.
```
Comment (EmilHvitfeldt) [inline tests/testthat/test-checks.R:184]: It took me a second to realize what happened here. A rewording would be nice. Maybe:

`errors informatively with needed package isn't installed`

File: R/checks.R [hunk 2]
```diff
@@ -201,9 +201,13 @@ check_installs <- function(x) {
   if (length(deps) > 0) {
     is_inst <- purrr::map_lgl(deps, is_installed)
     if (any(!is_inst)) {
-      rlang::abort(c("Some package installs are required: ",
-                     paste0("'", deps[!is_inst], "'", collapse = ", ")
-      ))
+      deps_not_installed <- deps[!is_inst]
+      deps_not_installed <- deps_not_installed[!duplicated(deps_not_installed)]
```
Comment (EmilHvitfeldt) [inline R/checks.R:205]:
```suggestion
      deps_not_installed <- unique(deps_not_installed)
```

Isn't this a fancy way to do `unique()`?

**Review by EmilHvitfeldt — APPROVED** [review]
looking good! two notes

File: R/checks.R (hunk 2 above)
Comment (simonpcouch) [reply R/checks.R]: Heard! [d6b958f](https://github.com/tidymodels/tune/pull/727/commits/d6b958f5c3a803b5ca4ef6af484ac74f377651db).

**Pull request merged**